### DISCUSSION TITLE
Welcome to OpenModScan Discussions!

### DISCUSSION THREAD

**Discussion by sanny32** [sanny32/OpenModScan#11] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by modmug** [comment]
Hi all, not sure if this is the correct place for my question, apologies if not. 
I am a novice at Modbus and OpenModScan. I need to send a command to a Modbus device that is connected to my laptop via an RS485 gateway to Ethernet. The command is 01030000000AC5CD 

I need to record the response from the device. When using OpenModScan I can see the laptop and device are connected and talking correctly. 

I just cannot find how to send the command and then receive the device response

Google AI tells me

AI Overview
The Modbus command 01 03 00 00 00 0A C5 CD is a request to a Modbus slave device, specifically for reading holding registers. Here is a breakdown of each component:
01: This is the Slave Address (or Unit ID). It indicates that the command is intended for the Modbus slave device with address 01.
03: This is the Function Code. 03 (or 0x03 in hexadecimal) signifies "Read Holding Registers." This function is used to read the contents of a contiguous block of holding registers in a remote device. 
00 00: This represents the Starting Register Address. In Modbus, register addresses are typically 0-indexed. 00 00 (hexadecimal) indicates that the request starts at the first holding register (address 0).
00 0A: This specifies the Number of Registers to Read. 00 0A (hexadecimal) translates to 10 in decimal, meaning the request is to read 10 consecutive holding registers starting from the specified address.
C5 CD: These are the CRC (Cyclic Redundancy Check) bytes. The CRC is a 16-bit checksum calculated over the preceding bytes of the message (01 03 00 00 00 0A). It is used by the slave device to verify the integrity of the received command and ensure no data corruption occurred during transmission. The slave will only process the command if the calculated CRC matches the received CRC.

**Comment by sanny32** [comment]
Hello @modmug 

> I need to record the response

Are you need record to file? Or you need view modbus request/response log?

If you want record modbus request/response log to file you can use 'Text Capture' option.
<img width="399" height="218" alt="image" src="https://github.com/user-attachments/assets/c8e4160d-201e-414f-988b-67a1e4d180b6" />

If you need view modbus log you must press 'Show Traffic' button on toolbar. 
<img width="59" height="51" alt="image" src="https://github.com/user-attachments/assets/e574d0e1-9041-4b77-8fc0-64e2edf0d929" />
You will see the traffic
<img width="1282" height="743" alt="image" src="https://github.com/user-attachments/assets/e58476cb-d85e-4a72-8697-0b27232c5331" />

**Comment by ZaroyRedrum** [comment]
I've been trying to learn the program and it's been very fun to learn. I downloaded and installed ModSim and Modscan. While I can configure both to send/receive data. I've been really curious about how to configure ModScan to act as either a master or slave to sending data to an Allen Bradley MicroLogix 1400. While I can set up two PLCs to traffic data utilizing MSG statements. I have not yet figured out how to configure ModScan and a PLC.

**Reply by sanny32** [reply]
See this video https://www.youtube.com/watch?v=7IsGY_7X7LM
